irida_staramr_results/amr_downloader.py

```diff
@@ -38,11 +39,23 @@ def download_all_results(irida_api, project_id, output_file_name, mode_append):
             result_files = irida_api.get_analysis_result_files(a["identifier"])
             data_frames = _files_to_data_frames(result_files)
             logging.info(f"Creating a file for analysis [{a['name']}]. ")
+            output_file_name = _get_output_file_name(a["createdDate"])
             _write_data_frames_to_excel(data_frames, output_file_name)
 
     logging.info(f"Download complete for project id [{project_id}].")
 
 
+def _get_output_file_name(timestamp):
```

Comment:
Should we pass in the user argument `output_file_name` into this function too?
The filename could be `<userinput>.YYYY-mm-ddTHH-MM-SS.xlsx` if a file name was provided.
You would have to check the filename the user provides to strip off the `.xlsx` before re-adding it if we want to do it this way.

In the other case, we could add a note that says the output name argument is only valid in `-a` mode, which is simpler codewise.

What are your thoughts on this?

Reply:
I think adding a little control over the file name is a good idea.

What I can do is set the `-o` as required but it will expect it without the `.xlsx` extension (but if the `.xlsx` is added anyway, that is fine, I will strip it off and re-add it).

Then overall, the output name format would be either:

`<userinput>-YYYY-mm-ddTHH-MM-SS.xlsx` or `<userinput>.xlsx`
     - I also remove the `.` between `<userinput>` and `YYYY-mm-ddTHH-MM-SS` because, although it ok, I read cause an error down the line.